File: leetcode/contest285/count-collisions-on-a-road.go
// https://leetcode.com/problems/count-collisions-on-a-road/
func countCollisions(d string) int {
	d = strings.TrimRight(d, "R")

	count, start, size := 0, 0, len(d)

	for start < size && d[start] == 'L' {
		start++
	}

	for start < size {
		if d[start] == 'R' {
			count += 1
		} else {
			if d[start] == 'L' {
				count += 1
			}
		}
		start++
	}
	return count
}

File: leetcode/contest285/count-collisions-on-a-road.cpp
// https://leetcode.com/problems/count-collisions-on-a-road/
int countCollisions(string d) {
  int s = d.size();
  int l = 0, r = s - 1, count = 0;
  while (l < s && d[l] == 'L') {
    l++;
  }
  while (r >= 0 && d[r] == 'R') {
    r--;
  }

  for (int i = l; i <= r; i++) {
    if (d[i] != 'S') {
      count++;
    }
  }
  return count;
}
